compiler/src/dmd/errors.d

```diff
@@ -483,6 +619,12 @@ extern (C++) void verrorReport(const ref Loc loc, const(char)* format, va_list a
                 {
                     info.headerColor = Classification.deprecation;
                     verrorPrint(format, ap, info);
+
+                    // Hook: Generate SARIF report only if the --sarif flag is set
+                    if (global.params.sarif)
+                    {
+                        generateSarifReport(info.loc, format, ap, info.kind);
+                    }
```

Comment:
Has this been checked against sarif readers such as github? What if there are multiple warnings or errors - is it happy with multiple json blocks or do all diagnostics need collating into one?

Reply:
Thanks for the feedback! 

I've noted this and will work on ensuring the SARIF output is properly collated for multiple diagnostics.

Reply:
I'll address handling multiple diagnostics in a separate PR, if that's okay. What do you think, @thewilsonator?